lib/player.js

```diff
@@ -918,6 +918,9 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     /** @private {boolean} */
     this.preloadDueAdManagerVideoEnded_ = false;
 
+    /** @private {!Array<HTMLTrackElement>} */
+    this.externalSrcEqualsTextTracks_ = [];
+
```

Comment:
All related logic is a bug in the Player, so create a separate PR to fix it (and we can cherry-pick it to another branches)